### PR TITLE
Added support for string and DateTimeInterface as Cookie::$expire

### PR DESCRIPTION
**Pull request by rhertogh**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | -
| Fixed issues  | #19917

The `\yii\httpclient\Response::parseCookie()` function sets the `Cookie::$expire` as string.
This PR broadens the accepted types for `Cookie::$expire` from `int` to `int|string|null`.

### CODE REVIEW

**Comment by bizley** [comment]
Ok, let's keep it as it seems that it's common thing to be prepared for other types than int. BUT since we are doing this let's allow DateTimeInterface as well.

File: framework/web/Cookie.php [hunk 1]
```diff
@@ -57,8 +57,8 @@ class Cookie extends \yii\base\BaseObject
      */
     public $domain = '';
     /**
-     * @var int the timestamp at which the cookie expires. This is the server timestamp.
-     * Defaults to 0, meaning "until the browser is closed".
+     * @var int|string|null the timestamp or date at which the cookie expires. This is the server timestamp.
```
Comment (bizley) [inline framework/web/Cookie.php:60]: Let's allow DateTimeInterface as well.

**Review by bizley — CHANGES_REQUESTED** [review]

File: framework/web/Cookie.php (hunk 1 above)
Comment (rhertogh) [reply framework/web/Cookie.php:60]: I'll add it 👍

**Review by bizley — APPROVED** [review]

**Review by Arhell — APPROVED** [review]

**Comment by bizley** [comment]
👍🏻

**Pull request merged**